lib/src/actions/user/add.rs

```diff
@@ -16,7 +17,14 @@ impl Action for UserAdd {
 
         let mut atoms: Vec<Step> = vec![];
 
-        atoms.append(&mut provider.add_user(&variant)?);
+        if variant.username.is_empty() {
+            return Ok(atoms)
+        }
+
+        match uzers::get_user_by_name(&variant.username) {
+            Some(_user) => debug!(message = "User already exists", username = ?variant.username),
+            None       => atoms.append(&mut provider.add_user(&variant)?)
+        }
```

Comment:
So we will probably need to guard this with conditional compilations so it doesn't get used on Windows, or compiled in to a windows binary, since the uzers crate only works on unix based system.

Reply:
Thanks @martintc, I've added a `#[cfg(unix)]` compilation guard.